src/debugging/netcore/NetCoreDebugHelper.ts

```diff
@@ -0,0 +1,259 @@
+            dockerOptions: {
+                containerNameToKill: containerName,
+                dockerServerReadyAction: dockerServerReadyAction,
+                removeContainerAfterDebug: debugConfiguration.removeContainerAfterDebug
+            },
+            pipeTransport: {
+                pipeProgram: 'docker',
+                // tslint:disable: no-invalid-template-strings
+                pipeArgs: ['exec', '-i', containerName, '${debuggerCommand}'],
+                pipeCwd: '${workspaceFolder}',
+                // tslint:enable: no-invalid-template-strings
+                debuggerPath: platformOS === 'Windows' ?
+                    'C:\\remote_debugger\\win7-x64\\latest\\vsdbg' :
+                    '/remote_debugger/vsdbg',
+                quoteArgs: false,
+            },
+            preLaunchTask: debugConfiguration.preLaunchTask,
+            sourceFileMap: debugConfiguration.sourceFileMap || {
+                '/app/Views': path.join(path.dirname(debugConfiguration.netCore.appProject), 'Views'),
+            }
+        };
+    }
+
+    public static getHostDebuggerPathBase(): string {
+        return path.join(os.homedir(), '.vsdbg');
+    }
+
+    private async inferAppOutput(helperOptions: NetCoreDebugOptions): Promise<string> {
+        return await this.netCoreProjectProvider.getTargetPath(helperOptions.appProject);
+    }
+
+    private async loadExternalInfo(folder: WorkspaceFolder, debugConfiguration: DockerDebugConfiguration): Promise<{ configureSsl: boolean, containerName: string, platformOS: PlatformOS }> {
+        const associatedTask = await getAssociatedDockerRunTask(debugConfiguration);
+
+        if (!associatedTask) {
+            throw new Error('Unable to find docker-run task associated with this project. Please make sure a docker-run task is defined for this project.');
+        }
+
+        return {
+            configureSsl: associatedTask.netCore && associatedTask.netCore.configureSsl !== undefined ? associatedTask.netCore.configureSsl : await NetCoreTaskHelper.inferSsl(folder, debugConfiguration.netCore),
+            containerName: associatedTask.dockerRun && associatedTask.dockerRun.containerName || `${await NetCoreTaskHelper.inferAppName(folder, debugConfiguration.netCore)}-dev`,
+            platformOS: associatedTask.dockerRun && associatedTask.dockerRun.os || 'Linux',
+        }
+    }
+
+    private async acquireDebuggers(platformOS: PlatformOS): Promise<void> {
+        if (platformOS === 'Windows') {
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'win7-x64');
+        } else {
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'linux-x64');
+            await this.vsDbgClientFactory().getVsDbgVersion('latest', 'linux-musl-x64');
+        }
+
+        const debuggerScriptPath = path.join(ext.context.asAbsolutePath('src/debugging/netcore'), 'vsdbg');
+        const destPath = path.join(NetCoreDebugHelper.getHostDebuggerPathBase(), 'vsdbg');

```

Comment:
>destPath [](start = 14, length = 8)

Does this path conflict with any shipped version of the VS tools?

Reply:
It shouldn't, VSCode does this currently with the structure of vsdbg directory:

- User Folder
  - .vsdbg
    - linux-x64/linux-musl-x64/win7-x64
      - vs2017u5 (now, 'latest')
    - GetVsDbg.ps1/.sh
    - vsdbg (this script)

VS uses 'vsdbg', not '.vsdbg' as the folder name. The fact they aren't sharing bits is a problem of its own, but maybe not a huge one if we expect users don't switch much between VS and VSCode.

Reply:
Note that older versions of VS use ".vsdbg"

Reply:
Good point. I checked out the old source, and it does this:

- User folder
  - .vsdbg
    - vs2017u5
      -linux-x64

So this shouldn't conflict, since we're doing .vsdbg\*\latest, instead of .vsdbg\vs2017u5\*.